plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/editor/autoedit/JSDocEditStrategy.java

```diff
@@ -0,0 +1,142 @@
+import org.eclipse.xtext.ui.editor.autoedit.CommandInfo;
+import org.eclipse.xtext.ui.editor.autoedit.MultiLineTerminalsEditStrategy;
+
+/**
+ * AutoEdit strategy which auto generates JSDoc comments.
+ */
+public class JSDocEditStrategy extends MultiLineTerminalsEditStrategy {
+	private static final String OPENING_STR = "/**";
+	private static final String INDENTATION_STR = " * ";
+	private static final String CLOSING_STR = " */";
+	private static final String PARAM_STR = "@param ";
+	private static final String RETURN_STR = "@return ";
+
+	/**
+	 * Call super with specified opening, closing and indentation string. JSDoc strategy is not nested.
+	 */
+	public JSDocEditStrategy() {
+		super(OPENING_STR, INDENTATION_STR, CLOSING_STR, false);
+	}
+
+	/**
+	 * Expects the cursor to be in the same line as the start terminal puts any text between start terminal and cursor
+	 * into a separate newline before the cursor. puts any text between cursor and end terminal into a separate newline
+	 * after the cursor. puts the closing terminal into a separate line at the end. adds a closing terminal if not
+	 * existent. If the next astElement is a method with parameters or return the JSDoc-tags will be added as an
+	 * addition.
+	 */
+	@Override
+	protected CommandInfo handleCursorInFirstLine(IDocument document, DocumentCommand command, IRegion startTerminal,
+			IRegion stopTerminal) throws BadLocationException {
+		CommandInfo newC = new CommandInfo();
+		List<String> returnTypeAndParameterNames = getReturnTypeAndParameterNames(document, startTerminal);
+		String paramString = "";
+		String returnString = "";
+		if ((returnTypeAndParameterNames.size() > 0) && returnTypeAndParameterNames.get(0).equals("return")) {
+			returnString = INDENTATION_STR + RETURN_STR + command.text;
+		}
+		if (returnTypeAndParameterNames.size() > 1) {
+			for (int i = 1; i < returnTypeAndParameterNames.size(); i += 1) {
+				paramString += command.text + INDENTATION_STR + PARAM_STR + returnTypeAndParameterNames.get(i);
+			}
+		}
+		newC.isChange = true;
+		newC.offset = command.offset;
+		newC.text += command.text + INDENTATION_STR;
+		newC.cursorOffset = command.offset + newC.text.length();
+		if (stopTerminal == null && atEndOfLineInput(document, command.offset)) {
+			newC.text += command.text + getRightTerminal();
+		}
+		if (stopTerminal != null && stopTerminal.getOffset() >= command.offset
+				&& util.isSameLine(document, stopTerminal.getOffset(), command.offset)) {
+			String string = document.get(command.offset, stopTerminal.getOffset() - command.offset);
+			if (string.trim().length() > 0)
+				newC.text += string.trim();
+			if (!(returnTypeAndParameterNames.size() == 0)) {
+				newC.text += paramString + command.text + returnString;
+			} else {
+				newC.text += command.text;
+			}
+
+			newC.length += string.length();
+		}
+		return newC;
+	}
+
+	/**
+	 * @returns List<String> otherwise null in case the document is not N4JSDocument or something breaks
```

Comment:
Add here: 

`The first item of the list (index 0) is the return type, the parameter names follow  at index 1.`